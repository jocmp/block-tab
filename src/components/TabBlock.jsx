import React from 'react';
import PropTypes from 'prop-types';

class TabBlock extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentIndex: 0
    };
    this.scrollToStart = this.scrollToStart.bind(this);
    this.scrollToEnd = this.scrollToEnd.bind(this);
  }

  componentWillMount() {
    this.props.fetchTabGroup('1');
  }

  scrollToStart() {
    this.carousel.scrollBy({ left: -this.carousel.clientWidth, behavior: 'smooth' });
  }

  scrollToEnd() {
    this.carousel.scrollBy({ left: this.carousel.clientWidth, behavior: 'smooth' });
  }

  render() {
    return (
      <div className="block-tab__container">
        <div className="block-tab__row">
          <div className="block-tab__wrapper">
            <div className="carousel__wrapper">
              <button className="carousel__button btn--left" onClick={ this.scrollToStart }>{ '<-' }</button>
              <div className="carousel" ref={ (carousel) => { this.carousel = carousel; } }>
                { this.props.tabGroup.tabs.map((tab) =>
                  <span key={ tab.id } className="carousel__tab">{ tab.title }</span>
                ) }
              </div>
              <button className="carousel-button btn--right" onClick={ this.scrollToEnd }>{ '->' }</button>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

TabBlock.propTypes = {
  fetchTabGroup: PropTypes.func.isRequired,
  tabGroup: PropTypes.shape({
    tabs: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      media: PropTypes.string
    }))
  })
};


export default TabBlock;