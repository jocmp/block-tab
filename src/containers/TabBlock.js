import { connect } from 'react-redux';
import TabBlock from '../components/TabBlock';
import { fetchTabGroup } from '../actions/BlockActions';

const mapStateToProps = ({ block }) => ({
  tabGroup: block
});

const mapDispatchToProps = (dispatch) => ({
  fetchTabGroup: (id) => dispatch(fetchTabGroup(id))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(TabBlock);