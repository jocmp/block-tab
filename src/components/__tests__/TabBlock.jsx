import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TabBlock from '../TabBlock';
import { tabGroup } from './fixtures';

describe('Quiz Feedback', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<TabBlock tabGroup={ tabGroup() } fetchTabGroup={ jest.fn() } />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});