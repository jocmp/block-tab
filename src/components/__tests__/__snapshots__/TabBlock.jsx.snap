// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quiz Feedback renders correctly 1`] = `
<div
  className="block-tab__container"
>
  <div
    className="block-tab__row"
  >
    <div
      className="block-tab__wrapper"
    >
      <Carousel
        onTabClick={[Function]}
        selectedTabId=""
        tabs={
          Array [
            Object {
              "description": "<p>Description here</p>",
              "id": "0001",
              "title": "Title 1",
            },
            Object {
              "description": "<p>Description here</p>",
              "id": "0010",
              "title": "Title 2",
            },
            Object {
              "description": "<p>Description here</p>",
              "id": "0101",
              "media": "fake://img.con",
              "title": "Title 3",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
